daemon/qrexec-client.c

```diff
@@ -280,18 +286,29 @@ static void send_exit_code(libvchan_t *vchan, int status)
     }
 }
 
-static void handle_input(libvchan_t *vchan)
+static void handle_input(libvchan_t *vchan, int data_protocol_version)
 {
-    char buf[MAX_DATA_CHUNK];
+    const size_t buf_len = max_data_chunk_size(data_protocol_version);
+    char *buf;
     int ret;
     size_t max_len;
     struct msg_header hdr;
 
-    max_len = libvchan_buffer_space(vchan)-sizeof(hdr);
-    if (max_len > sizeof(buf))
-        max_len = sizeof(buf);
+    max_len = libvchan_buffer_space(vchan);
+    if (max_len < sizeof(hdr))
+        return;
+    max_len -= sizeof(hdr);
+    if (max_len > buf_len)
+        max_len = buf_len;
     if (max_len == 0)
         return;
+
+    buf = malloc(buf_len);
```

Comment:
Is there a reason for allocating `buf_len` instead of `max_len`? I would call `buf_len`, `data_chunk_size` or something and only allocate `max_len`.

Reply:
Makes sense. Force-pushed update.